PyMca5/tests/ut/test_search.py

```diff
@@ -0,0 +1,234 @@
+import pytest
+
+from typing import List
+from unittest.mock import Mock, patch
+
+from pytestqt.qtbot import QtBot
+from tiled.client.base import BaseClient
+from tiled.queries import FullText, Key, Regex
+
+from PyMca5.PyMcaGui.io.QTiledSearch import QTiledSearchWidget
+from PyMca5.PyMcaGui.io.QTiledWidget import QTiledWidget
+from PyMca5.PyMcaGui.io.TiledRunSelector import TiledRunSelector
+
+
+def test_init(qtbot: QtBot, tiled_client_run_selector_model: TiledRunSelector):
+    """Can create a QTiledSearchWidget object."""
+    QTiledSearchWidget(model=tiled_client_run_selector_model)
+
+
+def test_render(qtbot: QtBot, tiled_client_run_selector_model: TiledRunSelector):
+    """Can render a QTiledSearchWidget window."""
+    search = QTiledSearchWidget(model=tiled_client_run_selector_model)
+    search.show()
+    qtbot.addWidget(search)
+
+
+def test_fulltext_checkbox_disables_key_field(qtbot: QtBot):
+    """Checked FullText checkbox should disable key and RegEx elements."""
+    search = QTiledSearchWidget(model=None)
+    search.show()
+    qtbot.addWidget(search)
+    # Key label/entry and regex checkbox should start enabled
+    assert all(
+        [
+            element.isEnabled()
+            for element in [
+                search.key_label,
+                search.key_entry,
+                search.regex_checkbox,
+            ]
+        ]
+    )
+    # full text hint should start invisible
+    assert search.full_text_hint.isVisible() == False
```

Comment:
Good suggestion from `ruff`
```suggestion
    assert not search.full_text_hint.isVisible()
```